src/shared/bg_public.h

```diff
@@ -1471,6 +1471,18 @@ void     AddFlag(int &flags, int flag);
 void     RemoveFlag(int &flags, int flag);
 void     ToggleFlag(int &flags, int flag);
 
+// true if first is lower than second argument
+// this function is like ` strcmp(first, second)` except constexpr.
+// outside of compile-time evaluations, prefer strcmp.
+constexpr int constexpr_strcmp(const char *first, const char *second)
+{
+       return *first == *second
```

Comment:
`foo ? bar ? 0 : 1 : 2` is unreadable to me. And here, you nested 3 ternary operators!

Imo, this would be more readable:

```
if ( *first == *second )
{
  return *first == '\0' ? 0 : constexpr_strcmp(first+1, second+1) );
}
return *first < *second ? -1 : 1;
```

You will notice that it does not even takes more lines...

Reply:
The thing is that this is two statements instead of one, and we need one (1) statement only in the body of a constexpr function

Reply:
(until we upgrade to a more recent C++ compiler, something like C++17)